snakesist/exist_client.py

```diff
@@ -235,26 +263,60 @@ def _delete_request(self, url: str) -> None:
     @staticmethod
     def _parse_item(node: etree._Element) -> QueryResultItem:
         return QueryResultItem(
-            node.attrib["absid"], node.attrib["nodeid"],
-            node.attrib["path"], TagNode(node[0], {})
+            node.attrib["absid"],
+            node.attrib["nodeid"],
+            node.attrib["path"],
+            TagNode(node[0], {}),
         )
 
     @property
     def base_url(self) -> str:
         """
         The base URL pointing to the eXist instance.
         """
-        return (
-            f"http://{self.user}:{self.password}@{self.host}:{self.port}"
-            f"/{self.prefix}"
-        )
+        return self._base_url
+
+    @property
+    def host(self):
+        """
+        The database hostname
+        """
+        return self.__connection_props.host
+
+    @property
+    def port(self):
+        """
+        The database port number
+        """
+        return self.__connection_props.port
+
+    @property
+    def user(self):
+        """
+        The user name used to connect to the database
+        """
+        return self.__connection_props.user
+
+    @property
+    def password(self):
+        """
+        The password used to connect to the database
+        """
+        return self.__connection_props.password
+
+    @property
+    def prefix(self):
+        """
+        The URL prefix of the database
+        """
+        return self.__connection_props.prefix
 
     @property
     def root_collection(self) -> str:
         """
         The configured root collection for database queries.
         """
-        return self._root_collection
+        return self.__connection_props.root_collection
```

Comment:
it isn't really a connection parameter. and hence could be changed directly. when sanity checks are required, using a getter/setter is needed. but for just a docstring, Sphinx recognizes those in the `__init__` method:

```python
self.root_collection = root_collection
""" The root collection … """
```

Reply:
i need to check if i got this correctly: are you suggesting to make `root_collection` an instance variable here and ditch the getter & setter?

Reply:
no, my only strong point is, that it's not a connection parameter.

maybe just keep the getter/setter, but add a sanity check … or, b/c i'd propose to do the sanity check by mangling the value through `pathlib.PurePosicPath`, maybe the value could even be stored as such. no opinions here, just ideas.